common/src/main/java/com/microsoft/identity/common/crypto/AndroidWrappedKeyLoaderFactory.kt

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.crypto
+
+import com.microsoft.identity.common.java.crypto.key.ISecretKeyLoader
+import com.microsoft.identity.common.java.flighting.CommonFlight
+import com.microsoft.identity.common.java.flighting.CommonFlightsManager
+
+/**
+ * Factory class for creating wrapped key loaders specific to the Android platform.
+ *
+ * This object is responsible for creating the appropriate implementation of [ISecretKeyLoader]
+ * based on feature flag. It abstracts away the details of which
+ * specific loader implementation should be used, allowing for runtime switching between
+ * different implementations without affecting client code.
+ */
+object AndroidWrappedKeyLoaderFactory {
```

Comment:
Rather than a keyloader factory, it looks like you need a factory for storage layer instead?

two ways to tackle this

1. the new storage layer can decrypt (read) with both old and new keyloader. it would only write with the new keyloader

2.  the storage re-encrypts its data with the new keyloader, and use the new keyloader to read/write.

Reply:
If we simply use the flight to switch here - it would lead to user's data loss (since they can't decrypt existing data)

Reply:
e.g. put the flight in AndroidBrokerEncryptionManagerWithKeyStoreWrappedKey's getKeyLoaderForEncryption() and getKeyLoaderForDecryption(), if you're going with the first route.

Reply:
Yeah, I’m planning to address this in the next PR, where I’ll be using strategy 1, see -> 
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/db8781af8f7746c3df4a51ee78f0fde2d82a661a/common/src/main/java/com/microsoft/identity/common/crypto/AndroidKeyStoreRsaKekManager.kt#L111

Instead of handling it at the storage layer as you mentioned, I’m just plugging in a new keyloader. This new keyloader will be easier to update if we need to introduce a new algorithm, whether for the key itself or the key-encryption-key.